test/inductor/test_torchinductor.py

```diff
@@ -1878,6 +1878,21 @@ def fn(a, b):
         b = make_tensor(10, 3, 352, 352, low=0, dtype=torch.float64, device=self.device)
         self.common(fn, (a, b), rtol=1e-4, atol=1e-5, check_lowp=False)
 
+    @config.patch(max_autotune_pointwise=True)
+    def test_split_cumsum_index(self):
+        # Split scan uses a workspace that needs to be zeroed before use.
+        # data[index] does indirect indexing that should catch issues if the
+        # workspace is not zeroed.
+        def fn(lengths, data):
+            offsets = torch.cumsum(lengths, 0)
+            return data[offsets]
```

Comment:
I don't think this test would ever run into issues with this.

The workspace stores partial sums that are shared between blocks. Since these don't actually change when re-running with the same arguments I don't think the result would ever be wrong. However, it would result in bad timing information as the partial results would all be available instantly and the blocks would never have to spin waiting.

tbh I'm not really sure how you could test that though. Perhaps add a verifier that runs the kernel and checks if any inputs were unexpectedly modified?

Reply:
If you comment out the changes in codegen/triton.py and run this test, the test does fail.

During autotuning different block sizes are attempted - when you switch from testing one block size to another, you'll fail device-side assertions.